tapir/coop/forms.py

```diff
@@ -315,78 +311,46 @@ def validate_share_owner(self):
                 ValidationError(_("This member is already resigned.")),
             )
 
-    def validate_choices(self):
-        coop_buys_shares_back = self.cleaned_data.get("coop_buys_shares_back")
-        willing_to_gift_shares_to_coop = self.cleaned_data.get(
-            "willing_to_gift_shares_to_coop"
-        )
-        transfering_shares_to = self.cleaned_data.get("transfering_shares_to")
-        only_one_choice_allowed_error_message = _("Please take only one choice.")
-
-        if coop_buys_shares_back and willing_to_gift_shares_to_coop:
-            self.add_error(
-                "coop_buys_shares_back", only_one_choice_allowed_error_message
-            )
-            self.add_error(
-                "willing_to_gift_shares_to_coop", only_one_choice_allowed_error_message
-            )
-        elif transfering_shares_to != None and (
-            coop_buys_shares_back or willing_to_gift_shares_to_coop
-        ):
-            self.add_error(
-                "transfering_shares_to", only_one_choice_allowed_error_message
-            )
-            if coop_buys_shares_back:
-                self.add_error(
-                    "coop_buys_shares_back", only_one_choice_allowed_error_message
-                )
-            elif willing_to_gift_shares_to_coop:
-                self.add_error(
-                    "willing_to_gift_shares_to_coop",
-                    only_one_choice_allowed_error_message,
-                )
+    def validate_transfer_choice(self):
+        resignation_type = self.cleaned_data.get("resignation_type")
+        transferring_shares_to = self.cleaned_data.get("transferring_shares_to")
 
         if (
-            transfering_shares_to == None
-            and not willing_to_gift_shares_to_coop
-            and not coop_buys_shares_back
+            resignation_type == MembershipResignation.ResignationType.TRANSFER
+            and transferring_shares_to is None
         ):
-            make_at_least_one_choice_error_message = "Please make a least one choice."
-            self.add_error(
-                "transfering_shares_to",
-                ValidationError(_(make_at_least_one_choice_error_message)),
-            )
-            self.add_error(
-                "willing_to_gift_shares_to_coop",
-                ValidationError(_(make_at_least_one_choice_error_message)),
-            )
             self.add_error(
-                "coop_buys_shares_back",
-                ValidationError(_(make_at_least_one_choice_error_message)),
```

Comment:
Here I wanted to be sure, that a least one choice is made. Don't you think this is necessary?

Reply:
Since the form shows a dropdown now, there is always one value selected, it's not possible to have nothing selected.